File: Data Structures & Algorithms/majority-element/submission-0.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        var count = 0;
        var res = nums[0];
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums[i] == res)
            {
                count++;
            }
            else
            {
                count--;
            }

            if (count == 0 && res != nums[i])
            {
                res = nums[i];
                count++;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/find-all-numbers-disappeared-in-an-array/submission-0.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<bool> res;
        int len = nums.size();
        res.reserve(len);
        for (int i = 0; i< len; i++)
        {
            res[i] = false;
        }
        for (int i : nums){
            res[i-1] = true;
        }
        vector<int> real;
        for(int i = 0; i< len; i++)
        {
            if(!res[i])
            {
                real.push_back(i+1);
            }
        }

        return real;
    }
};